prep_functions.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import pandas as pd
+from pathlib import Path
+from lxml import etree
+import cchardet as chardet
+
+
+class XML_Validator:
+
+    def __init__(self, xsd_path: str):
+        xmlschema_doc = etree.parse(
+            xsd_path)
+        self.xmlschema = etree.XMLSchema(
+            xmlschema_doc)
+
+    def validate(
+            self, xml_path: str) -> bool:
+        """_summary_ purpose checking for validated issues,
+        post download cycle on London Datastore,
+
+        :param xml_path: path to an xml file.
+        :type xml_path: str
+        :return:
+        :rtype: bool
+        """
+        xml_doc = etree.parse(xml_path)
+        result = self.xmlschema.validate(
+            xml_doc)
+
+        return result
+
+
+def file_encoding_audit(
+        data_folder: Path, ) -> pd.DataFrame:
+    """_summary_
```

Comment:
docstring,  function takes in a folder path object, it then uses the the cchardet library to fast detect the file encoding types,